broker-core/src/main/java/org/apache/qpid/server/virtualhostnode/AbstractVirtualHostNode.java

```diff
@@ -430,40 +430,42 @@ public PreferenceStore createPreferenceStore()
 
     protected final ConfiguredObjectRecord[] getInitialRecords() throws IOException
     {
-        ConfiguredObjectRecordConverter converter = new ConfiguredObjectRecordConverter(getModel());
+        final ConfiguredObjectRecordConverter converter = new ConfiguredObjectRecordConverter(getModel());
 
-        Collection<ConfiguredObjectRecord> records =
-                new ArrayList<>(converter.readFromJson(VirtualHost.class,this,getInitialConfigReader()));
-
-        if(!records.isEmpty())
+        try (final Reader initialConfigReader = getInitialConfigReader())
```

Comment:
Try-with-resource block is unnecessary here as reading code is closing the reader. If you really have to change to `try-with-resource`, it would be preferable to keep the `try` block small and only use it for reading the records. IMHO, that makes code more readable 
```
final Collection<ConfiguredObjectRecord> records;
try (final Reader initialConfigReader = getInitialConfigReader())
{
    records = new ArrayList<>(converter.readFromJson(VirtualHost.class, this, initialConfigReader));
}
```